### PR TITLE
fix typo.

### PR DESCRIPTION
**Pull request by johncming**
fix two typos, one of which is a bug.

### CODE REVIEW

**Comment by brian-brazil** [comment]
Do we have tests that should have caught this?

**Comment by johncming** [comment]
> Do we have tests that should have caught this?

There are no tests for this.  Add more tests for package`rulefmt` later is better.

This bug `0` is obvious. If I have more time and I will add more tests.

File: pkg/rulefmt/rulefmt.go [hunk 1]
```diff
@@ -98,7 +98,7 @@ func (g *RuleGroups) Validate(node ruleGroups) (errs []error) {
 		}
 	}
 
-	return errs
+	return
```
Comment (brian-brazil) [inline pkg/rulefmt/rulefmt.go]: There's no defer, so we shouldn't need a named return at all

Comment (bwplotka) [reply pkg/rulefmt/rulefmt.go:101]: Also we should never have `return` when there are arguments to return, this very prone to errors

Please see https://www.bwplotka.dev/2020/how-thanos-would-program-in-go/#use-named-return-parameters-carefully for rationales.

**Comment by roidelapluie** [comment]
@johncming Can you please address the comment and add a unit test that highlights the bug?

**Comment by johncming** [comment]
> @johncming Can you please address the comment and add a unit test that highlights the bug?

OK. 

Some changes: distinguishing between two situations of
error message for both setting and non-setting of 'alert' and 'record' at the same time.

File: pkg/rulefmt/rulefmt.go (hunk 1 above)
Comment (johncming) [reply pkg/rulefmt/rulefmt.go]: ok

Comment (brian-brazil) [reply pkg/rulefmt/rulefmt.go]: @bwplotka Are you happy with this now?

**Comment by roidelapluie** [comment]
Closing in favour of #12903. Thanks!

**Pull request closed without merging**